cmake/partition_manager.cmake

```diff
@@ -176,4 +176,25 @@ if(FIRST_BOILERPLATE_EXECUTION)
     set(ZEPHYR_RUNNER_CONFIG_KERNEL_HEX "${PROJECT_BINARY_DIR}/merged.hex"
       CACHE STRING "Path to merged image in Intel Hex format" FORCE)
   endif()
+
+  if (CONFIG_SECURE_BOOT AND CONFIG_BOOTLOADER_MCUBOOT)
+    # Create symbols for the offsets required for moving test update hex files
+    # to MCUBoots secondary slot. This is needed because objcopy does not support
+    # arithmetic expressions as argument (e.g. '0x100+0x200'), and all of the
+    # symbols used to generate the offset is only available as a generator
+    # expression when MCUBoots cmake code exectues.
+    math(EXPR s1_offset "${PM_S1_SIZE} + ${PM_MCUBOOT_PRIMARY_SIZE}")
```

Comment:
Why `PM_MCUBOOT_PRIMARY_SIZE` here?

Reply:
We need the offset from the start of S1 to the start of MCUBOOT_SECONDARY.
AFAICS this is the size of S1 + the whole size of MCUBOOT_PRIMARY (padding + app).